rest_authenticate/services/session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from odoo import fields
+from odoo.http import request, root
+from odoo.service import security
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+def _rotate_session(httprequest):
+    if httprequest.session.rotate:
+        root.session_store.delete(httprequest.session)
+        httprequest.session.sid = root.session_store.generate_key()
+        if httprequest.session.uid:
+            httprequest.session.session_token = security.compute_session_token(
+                httprequest.session, request.env
+            )
+        httprequest.session.modified = True
+
+
+class SessionAuthenticateService(Component):
+    _inherit = "base.rest.service"
+    _name = "session.authenticate.service"
+    _usage = "authenticate"
+    _collection = "session.rest.services"
+
+    @restapi.method([(["/"], "POST")])
+    def authenticate(self):
+        params = request.params
+        request.session.authenticate(params["db"], params["login"], params["password"])
+        result = request.env["ir.http"].session_info()
+        if request.session.rotate:
```

Comment:
this check is done already in `_rotate_session`

Reply:
Indeed, I remove it.